src/Kmd.Logic.Email.Client/EmailClient.cs

```diff
@@ -199,13 +199,38 @@ public async Task<IEnumerable<EmailProviderConfiguration>> GetAllProviderConfigu
                 {
                     case System.Net.HttpStatusCode.OK:
                         return configurationDetailsResponse.Body;
-
                     default:
                         throw new EmailException(configurationDetailsResponse?.Body?.ToString() ?? "Invalid configuration provided to access Email service");
                 }
             }
         }
 
+        /// <summary>
+        /// Get the information about email request.
+        /// </summary>
+        /// <param name="emailRequestDetails">Email request details.</param>
+        /// <returns>EmailDetailsResponse.</returns>
+        public async Task<EmailDetailsResponse> GetEmailDetails(EmailRequestDetails emailRequestDetails)
```

Comment:
Maybe we can rename it as `GetEmailRequestDetails` to match with the functionality.

Reply:
Done